packages/contracts/deploy/new/10_framework/01_ens_subdomains.ts

```diff
@@ -1,13 +1,45 @@
-import {DeployFunction} from 'hardhat-deploy/types';
-import {HardhatRuntimeEnvironment} from 'hardhat/types';
+import {ENSRegistry, ENSRegistry__factory} from '../../../typechain';
 import {
+  ENS_ADDRESSES,
   getContractAddress,
   getENSAddress,
   getPublicResolverAddress,
   registerSubnodeRecord,
   transferSubnodeChain,
 } from '../../helpers';
-import {ENSRegistry__factory} from '../../../typechain';
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
+import {DeployFunction} from 'hardhat-deploy/types';
+import {HardhatRuntimeEnvironment} from 'hardhat/types';
+
+async function registerAndTransferDomain(
+  ensRegistryContract: ENSRegistry,
+  managingDAOAddress: string,
+  domain: string,
+  node: string,
+  deployer: SignerWithAddress,
+  hre: HardhatRuntimeEnvironment,
+  ethers: any
+) {
+  let owner = await ensRegistryContract.owner(node);
+
+  if (owner !== managingDAOAddress && owner !== deployer.address) {
+    throw new Error(
+      `${domain} is not owned either by deployer: ${deployer.address} or management dao: ${managingDAOAddress}. 
+      Check if the domain is owned by ENS wrapper and if so, unwrap it from the ENS app.`
+    );
+  }
```

Comment:
Let's split this into two `if`s and throw two different errors

Reply:
I don't think that would be ideal. It could be the case that `owner` is not managing dao address but is deployer, in which case it shouldn't produce an error.